src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c

```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
+                this->dyna.actor.child->world.pos.z = this->dyna.actor.world.pos.z + sp28.z;
+                this->dyna.actor.child->flags &= ~1;
+            }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Movebg/func_8089E318.s")
+#endif
+
+void func_8089E650(BgMizuMovebg* this, GlobalContext* globalCtx) {
+    Vec3f waypoint;
+    f32 dist;
+    f32 dx;
+    f32 dy;
+    f32 dz;
+
+    this->dyna.actor.speedXZ = BgMizuMovebgGetSpeed(this->dyna.actor.params) * 0.1f;
+    func_8089E108(globalCtx->setupPathList, &waypoint, BgMizuMovebgGetPathId(this->dyna.actor.params),
+                  this->waypointId);
+    dist = Actor_WorldDistXYZToPoint(&this->dyna.actor, &waypoint);
+    if (dist < this->dyna.actor.speedXZ) {
+        this->dyna.actor.speedXZ = dist;
+    }
+    func_80035844(&this->dyna.actor.world.pos, &waypoint, &this->dyna.actor.world.rot, 1);
+    func_8002D97C(&this->dyna.actor);
+    dx = waypoint.x - this->dyna.actor.world.pos.x;
+    dy = waypoint.y - this->dyna.actor.world.pos.y;
+    dz = waypoint.z - this->dyna.actor.world.pos.z;
+    if (fabsf(dx) < 2.0f && fabsf(dy) < 2.0f && fabsf(dz) < 2.0f) {
+        this->waypointId++;
+        if (this->waypointId >= globalCtx->setupPathList[BgMizuMovebgGetPathId(this->dyna.actor.params)].count) {
+            this->waypointId = 0;
+            func_8089E108(globalCtx->setupPathList, &this->dyna.actor.world.pos,
+                          BgMizuMovebgGetPathId(this->dyna.actor.params), 0);
+        }
+    }
+    if (!(D_8089EE40 & 1) && BgMizuMovebgGetSpeed(this->dyna.actor.params) != 0) {
+        D_8089EE40 |= 1;
+        this->unk_17C |= 1;
+    }
+    if (this->unk_17C & 1) {
+        func_8002F948(&this->dyna.actor, NA_SE_EV_ROLL_STAND_2 - SFX_FLAG);
+    }
+}
+
+void BgMizuMovebg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuMovebg* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgMizuMovebg_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    BgMizuMovebg* this = THIS;
+    GlobalContext* globalCtx = (GlobalContext*)globalCtx2;
+    u32 frames;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_mizu_movebg.c", 754);
+    if (1) {}
+    frames = globalCtx->gameplayFrames;
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPSegment(POLY_OPA_DISP++, 0x08,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha1));
+
+    gSPSegment(POLY_OPA_DISP++, 0x09,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha2));
+
+    gSPSegment(POLY_OPA_DISP++, 0x0A,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha3));
```

Comment:
Do you really need the `0 * 0`?
```suggestion
    gSPSegment(POLY_OPA_DISP++, 0x08,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha1));

    gSPSegment(POLY_OPA_DISP++, 0x09,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha2));

    gSPSegment(POLY_OPA_DISP++, 0x0A,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha3));
```

Reply:
TwoTexScrolls being what they are, probably